spec/gp_update_metadata_source_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::GpUpdateMetadataSourceAction do
+  it 'updates any block in a given .po file with the values from the given sources' do
+    Dir.mktmpdir do |dir|
+      # 1: Create a dummy .po file to use as input.
+      output_path = File.join(dir, 'output.po')
+      dummy_text = <<~PO
+        msgctxt "v1.0-whats-new"
+        msgid "this value should change"
+        msgstr ""
+        msgctxt "release_note_0122"
+        msgid "previous version notes required to have current one added"
+        msgstr ""
+        msgctxt "key1"
+        msgid "this value should change"
+        msgstr ""
+        msgctxt "key2"
+        msgid "this value should change"
+        msgstr ""
+      PO
+      File.write(output_path, dummy_text)
```

Comment:
It was useful to have the file content inline in the test because I was jumping back and forth between it, the expected output, and the production code.

If we'll add more tests for new/modified code, it'll probably be best to have dedicated fixtures or methods doing this kind of work generation to keep the examples short and easy to glance